docs/quickstarts/nextjs.mdx

```diff
@@ -74,211 +75,130 @@ All Clerk hooks and components must be children of the [`<ClerkProvider>`](/docs
-      import type { AppProps } from "next/app";
-      function MyApp({ Component, pageProps }: AppProps) {
-        return (          
-          <ClerkProvider {...pageProps}>
-            <Component {...pageProps} />
-          </ClerkProvider>
-        );
-      }
-      export default MyApp;
-      ```
-
 </CodeBlockTabs>
 
 ### Add authentication to your app with middleware
 
 Now that Clerk is installed and mounted in your application, you can decide which pages are public and which should require authentication to access:
 
 1. Create a `middleware.ts` file at the root of your project, or in your `src/` directory if you have one.
-2. In your `middleware.ts`, export Clerk's [`clerkMiddleware`](/docs/references/nextjs/clerk-middleware) helper:
-    ```tsx filename="middleware.ts"
-    import { clerkMiddleware } from "@clerk/nextjs/server";
-
-    export default clerkMiddleware({});
-    ```
-3. With `clerkMiddleware()`, authentication will be enabled by default on all routes that your Next.js middleware runs on, blocking access to all signed out visitors. You can specify valid routes using Next.js's [`matcher`](https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher). Add the following code to your `middleware.ts` to protect your entire application:
-
-    ```tsx filename="middleware.ts" {15}
-    import { clerkMiddleware } from "@clerk/nextjs/server";
-
-    export default clerkMiddleware({
-      // Routes that can be accessed while signed out
-      publicRoutes: ['/anyone-can-visit-this-route'],
-      // Routes that can always be accessed, and have
-      // no authentication information
-      ignoredRoutes: ['/no-auth-in-this-route'],
-    });
-
-    export const config = {
-      // Protects all routes, including api/trpc.
-      // See https://clerk.com/docs/references/nextjs/clerk-middleware
-      // for more information about configuring your Middleware
-      matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
-    };
-    ```
-
-<Callout type="info">
-  See the [`authMiddleware()`](/docs/references/nextjs/auth-middleware#default-behaviour) docs to learn more about `publicRoutes` and `ignoredRoutes`.
-</Callout>
+2. In your middleware.ts, export Clerk's `clerkMiddleware()` helper:
+  ```tsx filename="middleware.ts"
+  import { clerkMiddleware } from "@clerk/nextjs/server";
+
+  export default clerkMiddleware();
+
+  export const config = {
+    matcher: [
+      '/((?!.*\\..*|_next).*)', // Don't run middleware on static files
+      '/', // Run middleware on index page
+      '/(api|trpc)(.*)'
+    ], // Run middleware on API routes
+  };
+  ```
+3. By default, `clerkMiddleware()` will not protect any routes. All routes are public and you must opt-in to protection for routes. See the [`clerkMiddleware()` docs](https://github.com/docs/references/nextjs/clerk-middleware) to learn how to require authentication for specific routes.
 
 ### Try accessing your app

```

Comment:
New problem.
Now that we are using clerkMiddleware, all pages are public.
So how does Account Portal work out of the box? The index page doesn't require sign-in, so it's not going to redirect the user to the AP sign-in page...